lib/analyze-component.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+const { walk } = require('./walk');
+
+const componentCache = new WeakMap();
+
+function getModuleScopedFunctions(root) {
+    return new Set(
+        root.body
+            .map((node) => {
+                if (node.type === 'FunctionDeclaration') {
+                    return (node.id && node.id.name) || '';
+                } else if (
+                    node.type === 'VariableDeclaration' &&
+                    node.declarations.length === 1 &&
+                    node.declarations[0].id.type === 'Identifier' &&
+                    ((node.declarations[0].init &&
+                        node.declarations[0].init.type === 'ArrowFunctionExpression') ||
+                        (node.declarations[0].init &&
+                            node.declarations[0].init.type === 'FunctionExpression'))
+                ) {
+                    return node.declarations[0].id.name;
+                }
+                return '';
+            })
+            .filter(Boolean),
+    );
+}
+
+function importsLightningElementClass(root) {
+    return root.body
+        .map(
+            (topLevelNode) =>
+                topLevelNode.type === 'ImportDeclaration' &&
+                topLevelNode.source.value === 'lwc' &&
+                topLevelNode.specifiers.find(
+                    (importSpecifier) =>
+                        'imported' in importSpecifier &&
+                        importSpecifier.imported.name === 'LightningElement',
+                ) &&
+                true,
+        )
+        .includes(true);
+}
+
+function getClassDeclaration(root) {
+    // TODO: iterate over method names to look for renderedCallback, connectedCallback, etc
```

Comment:
Can a WI be opened to track this? Or, if this isn't work doing, the comment can be removed. I don't have a strong opinion one way or another.